gulpfile.js

```diff
@@ -43,6 +43,8 @@ function readArgument(argumentName) {
         : null;
 }
 
+var hadTypescriptErrors = false;
```

Comment:
This is a big hack. `build` should fail in case of compilation errors, and `test` should not run. We should make the right fix here, which is fixing the `build` task if it does not report errors correctly.